dbt/adapters/athena/impl.py

```diff
@@ -267,3 +267,6 @@ def list_relations_without_caching(
                 )
 
         return relations
+
+    def valid_incremental_strategies(self):
```

Comment:
Is this needed? Seems not used.

Reply:
You're right, in the current PR it's not used. I'll remove it again.

We should refactor incremental materializations though (see the changes introduced in https://github.com/dbt-labs/dbt-core/pull/5359/files)

- Instead of doing if/else blocks in `incremental.sql` we should call this macro https://github.com/dbt-labs/dbt-snowflake/blob/65e282282a23578ac1083548356f55fba6959ea9/dbt/include/snowflake/macros/materializations/incremental.sql#L82 It will validate the strategy chosen based on this `valid_incremental_strategies` method like I implemented. 
- We should implement a default strategy with a macro like this: https://github.com/dbt-labs/dbt-snowflake/blob/65e282282a23578ac1083548356f55fba6959ea9/dbt/include/snowflake/macros/materializations/incremental.sql#L109-L111 and not use the `or default` workaround that I did in this PR. 
- For each strategy we support, we should implement a macro that generates the SQL: https://github.com/dbt-labs/dbt-core/blob/main/core/dbt/include/global_project/macros/materializations/models/incremental/strategies.sql
    - `athena__get_incremental_append_sql(arg_dict)`
    - `athena__get_incremental_insert_overwrite_sql(arg_dict)`
    - merge with iceberg in the future? See [snowflake merge macro](https://github.com/dbt-labs/dbt-snowflake/blob/65e282282a23578ac1083548356f55fba6959ea9/dbt/include/snowflake/macros/materializations/merge.sql)

Reply:
@Jrmyy have a look at this.